auth/user_mgt.go

```diff
@@ -0,0 +1,493 @@
+	params := map[string]interface{}{"maxResults": pageSize}
+	if pageToken != "" {
+		params["nextPageToken"] = pageToken
+	}
+	resp, err := it.client.makeUserRequest(it.ctx, "downloadAccount", params)
+	if err != nil {
+		// remove this line before submission ,see b/69406469
+		if pageToken != "" &&
+			strings.Contains(err.Error(), "\"code\": 400") &&
+			strings.Contains(err.Error(), "\"message\": \"INVALID_PAGE_SELECTION\"") {
+			return it.fetch(pageSize, "")
+		}
+		return "", err
+	}
+	var lur listUsersResponse
+	err = json.Unmarshal(resp, &lur)
+	if err != nil {
+		return "", err
+	}
+	for _, u := range lur.Users {
+		eu, err := makeExportedUser(u)
+		if err != nil {
+			return "", err
+		}
+		it.users = append(it.users, eu)
+	}
+	it.pageInfo.Token = lur.NextPage
+	return lur.NextPage, nil
+}
+
+func validateString(s *string, condition func(string) bool, message string) *string {
+	if s == nil || condition(*s) {
+		return nil
+	}
+	return &message
+}
+func validateStringLenGTE(s *string, name string, length int) *string {
+	return validateString(s, func(st string) bool { return len(st) >= length }, fmt.Sprintf("%s must be at least %d chars long", name, length))
+}
+
+func validateStringLenLTE(s *string, name string, length int) *string {
+	return validateString(s, func(st string) bool { return len(st) <= length }, fmt.Sprintf("%s must be at most %d chars long", name, length))
+}
+
+func validateCustomClaims(up *userParams) *string {
+	if up.CustomClaims == nil {
+		return nil
+	}
+	cc := up.CustomClaims
+	for _, key := range reservedClaims {
+		if _, ok := (*cc)[key]; ok {
+			return ptr.String(key + " is a reserved claim")
+		}
+	}
+	/*	if up.CustomAttributes == "" {
+		return ptr.String("attributes were not set, for non nil custom claims")
+	}*/
+	return validateStringLenLTE(up.CustomAttributes, "stringified JSON claims", 1000)
+}
+func validatePhoneNumber(phone *string) *string {
+	if phone == nil {
+		return nil
+	}
+	if len(*phone) == 0 {
+		return ptr.String("PhoneNumber cannot be empty")
+	}
+	if !strings.HasPrefix(*phone, "+") {
+		return ptr.String("PhoneNumber must begin with a +")
+	}
+	isAlphaNum := regexp.MustCompile(`[0-9A-Za-z]`).MatchString
+	if !isAlphaNum(*phone) {
+		return ptr.String("PhoneNumber must contain an alphanumeric character")
+	}
+	return nil
+}
+
+func validateEmail(email *string) *string {
+	if empty := validateStringLenGTE(email, "Email", 1); empty != nil {
+		return empty
+	}
+	if noAt := validateString(email,
+		func(s string) bool { return strings.Count(s, "@") == 1 },
+		"Email must contain exactly one '@' sign"); noAt != nil {
+		return noAt
+	}
+	return validateString(email,
+		func(s string) bool { return strings.Index(s, "@") > 0 && strings.LastIndex(s, "@") < (len(s)-1) },
+		"Email must have non empty account and domain")
+}
+
+func validateUID(uid *string) *string {
+
+	if tooLong := validateStringLenLTE(uid, "UID", 128); tooLong != nil {
```

Comment:
I would rather do the check here, instead of calling a set of helper functions. Code is more readable that way, and is not too long than it already is.

Reply:
ok is that what you meant ?